### PR TITLE
feat: supports collection compound indexes

### PR DESCRIPTION
**Pull request by yuanpli**
## Description

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/collections/config/types.ts [hunk 1]
```diff
@@ -240,6 +240,10 @@ export type CollectionConfig = {
     interface?: string
   }
   fields: Field[];
+  /**
+   * Custom collection indexes
```
Comment (DanRibbens) [inline src/collections/config/types.ts:244]: This comment can probably be the same as the docs for better explanation.

**Review by DanRibbens — CHANGES_REQUESTED** [review]
This is awesome! I love it. I gave some feedback items. If you have time great, if not I'll go ahead and make the changes for the next release. Well done!

File: test/collections-index/int.spec.ts
[the remark this answers is not in the document]
Comment (yuanpli) [reply test/collections-index/int.spec.ts:1]: Thanks for reviewing. I have made some changes according to your comments. Feel free to give me feedback.

File: docs/configuration/collections.mdx [hunk 2]
```diff
@@ -52,6 +53,12 @@ const Orders: CollectionConfig = {
 			relationTo: 'customers',
 			required: true,
 		}
+	],
+	indexes:[
```
[the remark this answers is not in the document]
Comment (yuanpli) [reply docs/configuration/collections.mdx:57]: Can you make this done before delivering this feature? Actually, I have no idea what example is better. In my opinion, the doc has a very clear explanation for this use case.

**Comment by ssyberg** [comment]
Amazing, I was just about to add a feature request for this!

**Review by DanRibbens — APPROVED** [review]
I pushed a few changes to get this over the finish. Thank you so much for the PR!

**Pull request merged**